packages/dcos-history/extra/history/server_util.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+import sys
+import threading
+
+
+from flask import Flask, Response
+from flask.ext.compress import Compress
+from history.statebuffer import BufferCollection, BufferUpdater
+
+
+compress = Compress()
+state_buffer = None
+log = logging.getLogger(__name__)
+add_headers_cb = None
+
+
+try:
+    import dcos_auth_python
+    log.info('dcos_auth_python module detected; applying settings')
+    global add_headers_cb
+    add_headers_cb = dcos_auth_python.get_auth_headers
+except ImportError:
+    log.info('no dcos_auth_python module detected; using defaults')
+
+
+def headers_cb():
+    """Callback method for providing headers per request
+
+    add_headers_cb is another callback providing headers (as a dict) to update the
+    defaults in this method. This method can be set by adding a dcos_auth_python package
+    with a get_auth_headers method
+    """
+    headers = {
+        "Access-Control-Allow-Credentials": "true",
+        "Access-Control-Allow-Headers": "accept, accept-charset, accept-encoding, " +
+                                        "accept-language, authorization, content-length, " +
+                                        "content-type, host, origin, proxy-connection, " +
+                                        "referer, user-agent, x-requested-with",
+        "Access-Control-Allow-Methods": "HEAD, GET, PUT, POST, PATCH, DELETE",
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Max-Age": "86400"}
+    if add_headers_cb:
+        headers.update(add_headers_cb())
+    return headers
+
+
+def create_app():
+    app = Flask(__name__)
+
+    logging.basicConfig(format='[%(levelname)s:%(asctime)s] %(message)s', level='INFO')
+
+    compress.init_app(app)
+
+    if 'HISTORY_BUFFER_DIR' not in os.environ:
+        sys.exit('HISTORY_BUFFER_DIR must be set!')
+
+    global state_buffer
+    state_buffer = BufferCollection(os.environ['HISTORY_BUFFER_DIR'])
+
+    def update():
```

Comment:
definitely looks like this update() function could be defined at the top level rather then nested.